### DISCUSSION TITLE
Sharing rules between grammars?

### DISCUSSION THREAD

**Discussion by claytonrcarter** [tree-sitter/tree-sitter#1360] [Q&A] [answered]
Is there a recommended best practice for sharing rules between grammars? For example, I'm tinkering with a phpdoc grammar, and it would be helpful to share some of the rules for types and such w/ the main php grammar. It appears that we can just do something like this:
```js
const otherGrammar = require('./path/to/other/grammar.js')

module.exports = grammar({
  rules: {
    my_rule: $ => otherGrammar.rules.their_rule
  }
});
```

As I said, this appears to work fine in practice, but I'm wondering if this is just coincidental and not recommended. 

Thank you!

**Comment by ahlinc** [comment, marked as answer]
You've mentioned a one of valid approaches if you want to share some rules between grammars that doesn't relate to each other. If you want to extend a grammar by some set of rules then you can look to the [tree-sitter-typescript](https://github.com/tree-sitter/tree-sitter-typescript/blob/83816f563c8d9d2f1b9c921206a7944d0c5904ad/common/define-grammar.js#L2) grammar and how it extends [tree-sitter-javascript ](https://github.com/tree-sitter/tree-sitter-javascript/blob/master/grammar.js)grammar. Also see the #645 issue for more info.
